zulipterminal/model.py

```diff
@@ -311,6 +313,19 @@ def get_messages(self, *,
             return True
         return False
 
+
+    def get_topics_in_all_streams(self) -> None:
+        """
+        Fetch all topics of a stream (Version 1)
+        """
+        # FIXME: Version 2: Fetch only last 'n' topics
+        #        for each stream.
+        for stream_id in self.stream_dict.keys():
+            response = self.client.get_stream_topics(stream_id)
+            if response['result'] == 'success':
+                self.index['topics'][stream_id] = [topic['name'] for \
+                      topic in response['topics']]
```

Comment:
The startup time for this is definitely too lengthy. With a carefully chosen method you should be able to run this in parallel with the other initial data-fetch methods in `_update_initial_data` (whose name could now be more appropriate). You might even split the list of stream_ids to work on them concurrently, as long as you can fetch the data back together safely.